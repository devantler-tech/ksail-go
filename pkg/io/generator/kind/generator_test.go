package kindgenerator_test

import (
	"testing"

	"github.com/devantler-tech/ksail-go/internal/testutils"
	generator "github.com/devantler-tech/ksail-go/pkg/io/generator/kind"
	generatortestutils "github.com/devantler-tech/ksail-go/pkg/io/generator/testutils"
	"github.com/gkampitakis/go-snaps/snaps"
	kindv1alpha4 "sigs.k8s.io/kind/pkg/apis/config/v1alpha4"
)

func TestMain(m *testing.M) { testutils.RunTestMainWithSnapshotCleanup(m) }

func TestGenerate(t *testing.T) {
	t.Parallel()

	gen := generator.NewKindGenerator()

	createCluster := func(_ string) *kindv1alpha4.Cluster {
		return &kindv1alpha4.Cluster{
			TypeMeta: kindv1alpha4.TypeMeta{
				APIVersion: "kind.x-k8s.io/v1alpha4",
				Kind:       "Cluster",
			},
		}
	}

	assertContent := func(t *testing.T, result, _ string) {
		t.Helper()
		snaps.MatchSnapshot(t, result)
	}

	generatortestutils.RunStandardGeneratorTests(t, gen, createCluster, "kind.yaml", assertContent)
}
