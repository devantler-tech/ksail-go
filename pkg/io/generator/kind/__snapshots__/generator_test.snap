
[TestGenerate/with_file - 1]
apiVersion: kind.x-k8s.io/v1alpha4
kind: Cluster
networking: {}

---

[TestGenerate/with_force_overwrite - 1]
apiVersion: kind.x-k8s.io/v1alpha4
kind: Cluster
networking: {}

---

[TestGenerate/without_file - 1]
apiVersion: kind.x-k8s.io/v1alpha4
kind: Cluster
networking: {}

---
